server/user_db.js

```diff
@@ -303,3 +303,66 @@ exports.resetPassword = function(password, email, user_id, callback) {
     }
   });
 };
+
+exports.getUserFollowedEventsByUid = (user_id, callback) => {
+  if (!user_id) {
+    return callback(null, '[]');
+  }
+  let query = 'SELECT followed_event FROM user_profile WHERE uid = $1';
+  db.query(query, [user_id], (err, result) => {
+    if (err) {
+      callback(err);
+    } else {
+      callback(null, result.rows[0]['followed_event']);
```

Comment:
I am getting this error while i am logged in, make sure no error when there are no followed events exist
"user_db.js:316
      callback(null, result.rows[0]['followed_event']);
TypeError: Cannot read property 'followed_event' of undefined"

Reply:
I have found the problem. A newly registered user has no profile created in the user_profile table. I have created a new user_api to check the existence of the profile.